pkg/processor/runtime/python/py/_nuclio_wrapper.py

```diff
@@ -193,6 +203,20 @@ async def _initialize_context(self):
                 self._logger.error('Exception raised while running init_context')
                 raise
 
+    def _register_to_signal(self):
+        signal.signal(signal.SIGTERM, self._on_sigterm)
+
+    def _on_sigterm(self, signal_number, frame):
+        self._logger.debug_with('Received signal, calling termination callback', signal=signal_number)
+
+        # cancel current event handling task if exists
+        if self._current_event_handling_task is not None:
+            self._current_event_handling_task.cancel()
+
+        # TODO: send a control message to the processor after this line,
+        # to indicate that the termination handler has finished, and the processor can exit early
+        self._platform._on_signal()
```

Comment:
And we should only signal completion once the user callback returns (I guess this hasn't made it into the PR yet?).

Reply:
It did but I reverted it after speaking with @liranbg.
Let's discuss it offline.